Java-FlightServer/src/main/java/ibm/com/example/ExampleFlightServer.java

```diff
@@ -0,0 +1,156 @@
+        this.allocator = incomingAllocator.newChildAllocator("flight-server", 0, Long.MAX_VALUE);
+        this.location = location;
+        Producer producer = new Producer();
+        this.flightServer = FlightServer.builder(this.allocator, location, producer).build();
+        this.isNonBlocking = isNonBlocking;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public void start() throws IOException {
+        flightServer.start();
+    }
+
+    public void awaitTermination() throws InterruptedException {
+        flightServer.awaitTermination();
+    }
+
+    /**
+     *  Main method starts the server listening to localhost:12233.
+     */
+    public static void main(String[] args) throws Exception {
+        final BufferAllocator a = new RootAllocator(Long.MAX_VALUE);
+        final ExampleFlightServer efs = new ExampleFlightServer(a, Location.forGrpcInsecure("localhost", 12233));
+        efs.start();
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                System.out.println("\nExiting...");
+                AutoCloseables.close(efs, a);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }));
+        efs.awaitTermination();
+    }
+
+    private final class Producer extends NoOpFlightProducer {
+        private final int RecordsPerBatch = 1024*1024;
+        private final VectorSchemaRoot constVectorSchemaRoot = getConstVectorSchemaRoot();
+
+        private VectorSchemaRoot getConstVectorSchemaRoot() {
+            final Schema schema = new Schema(ImmutableList.of(
+                    Field.nullable("a", MinorType.BIGINT.getType()),
+                    Field.nullable("b", MinorType.BIGINT.getType()),
+                    Field.nullable("c", MinorType.BIGINT.getType()),
+                    Field.nullable("d", MinorType.BIGINT.getType())
+            ));
+            //Schema schema = Schema.deserialize(ByteBuffer.wrap(perf.getSchema().toByteArray()));
+            VectorSchemaRoot root = VectorSchemaRoot.create(schema, allocator);
+            //root.allocateNew();
+            BigIntVector a = (BigIntVector) root.getVector("a");
+            BigIntVector b = (BigIntVector) root.getVector("b");
+            BigIntVector c = (BigIntVector) root.getVector("c");
+            BigIntVector d = (BigIntVector) root.getVector("d");
+
+            for (int j = 0; j < this.RecordsPerBatch; j++) {
+                a.setSafe(j, j);
+                b.setSafe(j, j);
+                c.setSafe(j, j);
+                d.setSafe(j, j);
+            }
+            root.setRowCount(this.RecordsPerBatch);
+            return root;
+        }
+
+        @Override
+        public void getStream(CallContext context, Ticket ticket,
+                              ServerStreamListener listener) {
+            final Runnable loadData = () -> {
+                listener.setUseZeroCopy(true);
```

Comment:
Seems to be improper use because it states that you can't use the vectors again after the first putNext call

Reply:
@roee88 : need to investigate. When I change to `false`, I quickly crash with:
`java.lang.OutOfMemoryError: Direct buffer memory`